### DISCUSSION TITLE
SelectStatementGrouping' object has no attribute 'is_derived_from' on join

### DISCUSSION THREAD

**Discussion by liverpool1026** [sqlalchemy/sqlalchemy#10135] [Usage Questions]
We are migrating from 1.3 -> 1.4.

One of our query is working in 1.3.

We have a joined inheritance table structure like this

class Group:
      ...

class ChildGroup(Group):
    ....

class GrandchildGroup(ChildGroup):
   ....

and a mapping table which correspond to each one of them

class GroupMapping:
     group : sa.orm.RelationshipProperty = relationship("Group", viewonly=True)

class ChildGroupMapping(GroupMapping):
    child_group= synonym("group")

class GrandchildGroupMapping(ChildGroupMapping):
    grandchild_group= synonym("group")

And the query is like this

```
sq = query.subquery()

q = data_set.query(
      ...
)
.select_from(sq)
.join(
    GrandchildGroupMapping, ..
)
.join(
**GrandchildGroup, GrandchildGroupMapping.grandchild_group**
)

```
The line where we joined using the relationship to join no longer works on 1.4

And I am getting this error, can anyone point me to whether this is a bug or I just did something stupid.

![image](https://github.com/sqlalchemy/sqlalchemy/assets/29122286/1301d65c-466d-492d-a54c-7f6d7d1fa921)

Thanks

**Comment by zzzeek** [comment]
Well definitely a bug, you shouldn't get attribute error even if you were using a construct in the wrong way 

On Sun, Jul 23, 2023, at 6:44 PM, Kevin Hwa wrote:
> 
> 
> We are migrating from 1.3 -> 1.4.
> 
> One of our query is working in 1.3.
> 
> We have a joined inheritance table structure like this
> 
> 
> class Group:
> ...
> 
> 
> class ChildGroup(Group):
> ....
> 
> 
> class GrandchildGroup(ChildGroup):
> ....
> 
> and a mapping table which correspond to each one of them
> 
> 
> class GroupMapping:
> group : sa.orm.RelationshipProperty = relationship("Group", viewonly=True)
> 
> 
> class ChildGroupMapping(GroupMapping):
> child_group= synonym("group")
> 
> 
> class GrandchildGroupMapping(ChildGroupMapping):
> grandchild_group= synonym("group")
> 
> And the query is like this
> 
> `sq = query.subquery()
> 
> q = data_set.query(
>       ...
> )
> .select_from(sq)
> .join(
>     GrandchildGroupMapping, ..
> )
> .join(
> **GrandchildGroup, GrandchildGroupMapping.grandchild_group**
> )
> 
`

**Comment by zzzeek** [comment]
OK trying to come up with a test, it's not clear if Group etc are mapped, and how XYZMapping relates to XYZ ? Can you please share a minimal complete mapping and demonstration?   (see [MCVE](http://stackoverflow.com/help/mcve))

**Reply by liverpool1026** [reply]
Yeah, I am still working on it to make a MCVE. Seems like it is only happening for an unique case. Still trying to see if I can reproduce it
